### ISSUE TITLE
Regroup the options from the view menu of the 3D viewport.

### ISSUE THREAD

**Issue by Mautar55** [godotengine/godot-proposals#11256]
### Describe the project you are working on

I'm working on some prototypes, however this is a general usability improvement. I'm working on it already.

### Describe the problem or limitation you are having in your project

There are too many options into the one menu of the 3D viewport. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Separate the view menu into three menus, so the tools and modes are easier to find. I would choose some new icons too.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

The new menus would be:

- The one showing the camera mode (Perspective by default). One off tools related to the camera go here too.
- The one showing the Display options, would show the selected mode. I also would like to change "Normal" by "Final Render". 
- Tools, which would contain all the options that toggle editor overlays. Also the ones that are to be used actively (the two audio related and cinematic preview).

Current:
![Image](https://github.com/user-attachments/assets/8e98c392-c3f0-4798-8d91-1f784afe4a5c)

Proposal:
![Image](https://github.com/user-attachments/assets/60237181-e13b-4883-a788-cb6542d34757)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No

### Is there a reason why this should be core and not an add-on in the asset library?

It is a core part of the editor.

**Comment by Calinou** [comment]
I would name the "Final Render" button with the name of the current debug draw mode instead (or "Normal" if not using a debug draw mode).

**Comment by Mautar55** [comment]
> I would name the "Final Render" button with the name of the current debug draw mode instead (or "Normal" if not using a debug draw mode).

That's what I meant. But also stop calling that option Normal, since can be confused with a normal buffer.

**Comment by Calinou** [comment]
> But also stop calling that option Normal, since can be confused with a normal buffer.

I'm not sure if "Final Render" would be what I go for, it feels strangely elaborate. Maybe just "Render"?

**Comment by Mautar55** [comment]
> I'm not sure if "Final Render" would be what I go for, it feels strangely elaborate. Maybe just "Render"?

Seems better.

> Remember that this adds 2 new buttons which will take more space on the viewport, and you can have up to 4 viewports in a split arrangement at the same time. We need to ensure the new buttons don't take too much additional space on small displays.

Yes. This will be an issue with labels such as "Reflection Probe Cluster" and "Directional Shadow Map" . Tools could be reduced to an icon only button, as it is quite generic. 
 ![Image](https://github.com/user-attachments/assets/fbebbdb6-0438-48d1-aad7-5154594dd37d)
Buttons could also be stacked if there isn't enough horizontal space. Not much better still...

**Comment by Calinou** [comment]
> This will be an issue with labels such as "Reflection Probe Cluster" and "Directional Shadow Map" . 

We could use abbreviations for specific debug draw modes, but maybe it's not needed if these are rarely used. I was mostly concerned about the length the "default" case (which is, not using a debug draw mode).

**Comment by Mautar55** [comment]
> We could use abbreviations for specific debug draw modes, but it may not be needed if these are rarely used. I was mostly concerned about the length the "default" case (which is, not using a debug draw mode).

Ok then, this is similar to what would be a worst-case scenario:

![Image](https://github.com/user-attachments/assets/65dd94a4-0249-4abf-bade-0b7518d8d4b6)

**Comment by Jesusemora** [comment]
Can we have buttons inside of bars and not floating in the viewport where we have to click? That's a serious problem with modern blender.
As for the menues, I don't see the point, it's not something we click all the time, so it's fine all in one button.
I change perspectives with the keyboard anyways. I only use that menu to look at SDFGI cascades, AO, etc.

**Comment by tetrapod00** [comment]
Making this massive menu smaller is long overdue.

Converting it into multiple menus like this seems like a pretty good solution, and as mentioned is in line with what other 3D editors do. Adding more clutter to the viewport itself is a downside, but seems worth it. Still worth discussing other solutions.

I didn't see it discussed in this thread yet, but putting it out there: I'm skeptical of any changes that *only* increase the nesting of submenus within this menu. It might be a good solution for some of the clutter in this menu, but probably isn't a solution for *all* of it.

**Comment by Mautar55** [comment]
> Can we have buttons inside of bars and not floating in the viewport where we have to click? That's a serious problem with modern blender. 

Can you elaborate on the issue of floating buttons you have in Blender? The only ones I can think of are the tools at the left. Is it because of visibility or missclick?

**Comment by Jesusemora** [comment]
> > Can we have buttons inside of bars and not floating in the viewport where we have to click? That's a serious problem with modern blender.
> 
> Can you elaborate on the issue of floating buttons you have in Blender? The only ones I can think of are the tools at the left. Is it because of visibility or missclick?

godot doesn't have the problems of blender (for now). I will try to explain in more detail what I mean.

this was the UI for bender 2.79, it was simple, discrete, putting the 3D model first and UI second.
![Image](https://github.com/user-attachments/assets/9bb64e49-d415-4281-9365-fccd64ebc7f4)
when using a tool like bevel, the options appeared in the sidetab, most of the tools and options were in these tabs, similar to the godot inspector, and others were in toolbars:
![Image](https://github.com/user-attachments/assets/b9e326ae-6fc1-4fac-9d69-6a7b77fbc74a)

with the change to blender 3, they made all the bars into floating buttons. these have two problems, these being miss-clicks, and covering of content. when working with a complex scene, you move the camera to the point where you see all that you need and you can select faces or objects one by one. but these floating buttons take space away from the viewport, forcing you to move the camera constantly, which in some narrow indoor scenes is not always possible.

![Image](https://github.com/user-attachments/assets/1b1f843e-aaf5-4b7a-ae5b-27ae04e952f2)

there are also tools on the other side.
in this example, they make it difficult to click on the gizmo to move the face:
![Image](https://github.com/user-attachments/assets/ac381e91-d80e-45c5-9d88-b24d8997e5e4)

and in this example, there's a transparent bar, but it's just cosmetic. clicking on it can accidentally move the cursor:
![Image](https://github.com/user-attachments/assets/778f3ae0-7f81-4a02-9b2e-13b364830dc3)


None of these are problems that godot has (yet). let's not go in that direction, let's not just blindly copy what "the 3D apps are doing", because it's not always good.
it might "look better", but it harms usability.

I think, if this ends up being changed, it would be best to have a single horizontal bar with the buttons and menus instead of a bunch of floating items.
the buttons should be called:
View, Display and Gizmos, or just View and Display.

or if possible, move them to the toolbar:
![Image](https://github.com/user-attachments/assets/e0a48a72-718d-4d72-a260-bd06d4380918)
![Image](https://github.com/user-attachments/assets/7e95f64b-272f-4990-9995-b950b670b4f6)

**Comment by Mautar55** [comment]
> godot doesn't have the problems of blender (for now). I will try to explain in more detail what I mean.

Yes. I get annoyed by that too time to time. I'll take it into account.

**Comment by DevPoodle** [comment]
While these are being reorganized, I think we should consider moving the "Align Transform with View" and "Align Rotation with View" buttons. All of the other options are ways to change the editor view, but these are just types of transformations. I personally think having it in the "Transform" tab of the toolbar would make more sense and would make it easier to find for new users. I could easily be wrong though, I just remember struggling to find it when I started using the engine.

![Image](https://github.com/user-attachments/assets/ea1c406d-91b7-4820-8453-fbf2b86afc77)

**Comment by Mautar55** [comment]
> While these are being reorganized, I think we should consider moving the "Align Transform with View" and "Align Rotation with View" buttons. All of the other options are ways to change the editor view, but these are just types of transformations. I personally think having it in the "Transform" tab of the toolbar would make more sense and would make it easier to find for new users. I could easily be wrong though, I just remember struggling to find it when I started using the engine.
> 
> ![Image](https://github.com/user-attachments/assets/ea1c406d-91b7-4820-8453-fbf2b86afc77)

How would you know to which view align if the viewport is split? In such cases there's only one top bar for all of them.

**Comment by DevPoodle** [comment]
> How would you know to which view align if the viewport is split? In such cases there's only one top bar for all of them.

Ah, that's a good point. I guess it's fine where it is then.

**Comment by vPumpking** [comment]
maybe It could be more of the same menu divided into several submenus like what we find in any wordprocessor or advanced software:

![Image](https://github.com/user-attachments/assets/4c61d006-499e-47cb-9de9-ce8dbf8cd653)

![Image](https://github.com/user-attachments/assets/e1c170ce-db0a-4191-bd11-72dfcdb01021)

![Image](https://github.com/user-attachments/assets/36d72a82-af79-479a-b1d2-42acd0ea0017)

(this is ugly but you get the idea)

**Comment by Nallebeorn** [comment]
> > maybe It could be more of the same menu divided into several submenus like what we find in any wordprocessor or advanced software:
> 
> This means we'd have "sub-sub-menus" for things like advanced display modes, which is generally considered poor UX.

With the display modes in their own menu, would the advanced display modes still need to be hidden under a sub-menu? There should be space for all modes in a single flat list at that point. (I personally still think separate buttons would be better though, sub-menus often feel clunky to use)

Another thing -- many of the first entries in the view menu relate to and in several cases duplicate functionality of the view gizmo in the top-right corner (i.e. all the "Top View" etc. options). Maybe those options could be accessed with an icon/button next to that gizmo? It would take up less space than a new labelled floating button, and logically group related functionality.
